unit_tests/speech/test_client.py

```diff
@@ -0,0 +1,111 @@
+# Copyright 2016 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+
+
+class TestClient(unittest.TestCase):
+
+    def _getTargetClass(self):
+        from google.cloud.speech.client import Client
+        return Client
+
+    def _makeOne(self, *args, **kw):
+        return self._getTargetClass()(*args, **kw)
+
+    def test_ctor(self):
+        from google.cloud.speech.connection import Connection
+
+        creds = _Credentials()
+        http = object()
+        client = self._makeOne(credentials=creds, http=http)
+        self.assertTrue(isinstance(client.connection, Connection))
+        self.assertTrue(client.connection.credentials is creds)
+        self.assertTrue(client.connection.http is http)
+
+    def test_syncrecognize(self):
+        from google.cloud.speech.client import Encoding
+
+        creds = _Credentials(object())
+        http = _Http(headers={},
+                     content={
+                         "results": [
+                             {
+                                 "alternatives": [
+                                     {"transcript": "hello",
+                                      "confidence": 0.784919}
+                                     ]
+                             }
+                         ]
+                     })
+        client = self._makeOne(credentials=creds, http=http)
+
+        uri = 'gs://a-bucket/rec_sample.flac'
+        encoding = Encoding.FLAC
+        sample_rate = 16000
+        hints = ["test"]
+        speechrecognition_result = client.syncrecognize(None, uri, encoding,
+                                                        sample_rate,
+                                                        max_alternatives=2,
+                                                        speech_context=hints)
+        self.assertEqual(speechrecognition_result[0]["transcript"], 'hello')
```

Comment:
Please split these two into separate tests (one with the URI, one with the content).  Add assertions to each that the right method was called, and that the expected `audio` and `configuration` mappings were passed as the payload.  For one of the two cases, please demonstrate non-default values for the optional parameters (and verify that they get added correctly to the `configuration` mapping).